sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -118,6 +118,12 @@ def __init__(self, operation_name, spec, counter_factory, state_sampler):
     self.counter_factory = counter_factory
     self.consumers = collections.defaultdict(list)
 
+    # These are overwritten in the legacy harness.
+    self.step_name = operation_name
+    self.metrics_container = MetricsContainer(self.step_name)
+    self.scoped_metrics_container = ScopedMetricsContainer(
+        self.metrics_container)
```

Comment:
Nice. This bothered me for a while : )
Can you then remove the setting of these variables in operations.py:407, and operations.py:615-617?

Reply:
Unfortunately I can't remove them from operations.py as the step and stage names can't be unified in that case (with the legacy harness). But this code will all be going away soon.